drltrace/drltrace.c

```diff
@@ -104,6 +105,119 @@ static app_pc exe_start;
 /* runtest.cmake assumes this is the prefix, so update both when changing it */
 #define STDERR_PREFIX "~~~~ "
 
+#define MAX_LIBCALL_NAME 255
+
+/* XXX: The functions print_simple_value and print_arg were taken from drstrace.
+ * It would be better to move them in drsyscall and import in drstrace and here.
+ */
+static void
+print_simple_value(drsys_arg_t *arg, bool leading_zeroes)
+{
+    bool pointer = !TEST(DRSYS_PARAM_INLINED, arg->mode);
+    dr_fprintf(outf, pointer ? PFX : (leading_zeroes ? PFX : PIFX), arg->value);
+    if (pointer && ((arg->pre && TEST(DRSYS_PARAM_IN, arg->mode)) ||
+                    (!arg->pre && TEST(DRSYS_PARAM_OUT, arg->mode)))) {
+        ptr_uint_t deref = 0;
+        ASSERT(arg->size <= sizeof(deref), "too-big simple type");
+        /* We assume little-endian */
+        if (dr_safe_read((void *)arg->value, arg->size, &deref, NULL))
+            dr_fprintf(outf, (leading_zeroes ? " => " PFX : " => " PIFX), deref);
+    }
+}
+
+static void
+print_arg(drsys_arg_t *arg)
+{
+    dr_fprintf(outf, "\n    arg %d: ", arg->ordinal);
```

Comment:
In case of known I think we have pretty good scheme of printing. One line would be a bit complicated to read in case of known arguments while for unknown arguments it looks ok. I would prefer to leave current scheme for now...